### PR TITLE
feat: Add some benchmarks to the tests repo

### PR DESCRIPTION
**Pull request by rw0x0**

### CODE REVIEW

File: circom-mpc-vm/src/accelerator.rs [hunk 1]
```diff
@@ -6,11 +6,11 @@ use mpc_core::traits::CircomWitnessExtensionProtocol;
...
-        &mut C,
-        &[<C as CircomWitnessExtensionProtocol<<P as Pairing>::ScalarField>>::VmType],
-    ) -> eyre::Result<
-        Vec<<C as CircomWitnessExtensionProtocol<<P as Pairing>::ScalarField>>::VmType>,
-    >,
+            &mut C,
+            &[<C as CircomWitnessExtensionProtocol<<P as Pairing>::ScalarField>>::VmType],
+        ) -> eyre::Result<
+            Vec<<C as CircomWitnessExtensionProtocol<<P as Pairing>::ScalarField>>::VmType>,
+        > + Send,
```
Comment (dkales) [inline circom-mpc-vm/src/accelerator.rs:13]: can we make this commit a separate PR?

File: collaborative-groth16/src/groth16.rs [hunk 2]
```diff
@@ -116,8 +116,20 @@ where
     pub witness: FieldShareVec<T, P>,
 }
 
+impl<T, P: Pairing> Clone for SharedWitness<T, P>
```
Comment (dkales) [inline collaborative-groth16/src/groth16.rs:119]: a comment on why this is needed (deriving clone would add bounds on T,P as well which are not needed)

File: collaborative-groth16/src/groth16.rs [hunk 3]
```diff
@@ -138,6 +150,18 @@ where
     pub shared_inputs: BTreeMap<String, T::FieldShareVec>,
 }
 
+impl<T, P: Pairing> Clone for SharedInput<T, P>
```
Comment (dkales) [inline collaborative-groth16/src/groth16.rs:153]: same here

**Review by dkales — CHANGES_REQUESTED** [review]

File: collaborative-groth16/src/groth16.rs (hunk 3 above)
Comment (rw0x0) [reply collaborative-groth16/src/groth16.rs:155]: done (in other PR)

File: collaborative-groth16/src/groth16.rs (hunk 2 above)
Comment (rw0x0) [reply collaborative-groth16/src/groth16.rs:120]: done (in other PR)

File: circom-mpc-vm/src/accelerator.rs (hunk 1 above)
Comment (rw0x0) [reply circom-mpc-vm/src/accelerator.rs:13]: done (in other PR)

**Review by dkales — APPROVED** [review]
